configuration-maven-extension/build.gradle.kts

```diff
@@ -13,17 +15,21 @@ group = "com.gradle"
 description = "Maven extension to configure injected Gradle Enterprise Maven extension"
 version = "1.0.0"
 
-val gradleEnterpriseMavenExtensionVersion: String by (gradle as ExtensionAware).extra
+val develocityMavenExtensionVersion: String by (gradle as ExtensionAware).extra
 
+// only necessary as long as we use a rc
 repositories {
     mavenCentral()
+    maven {
+        url = URI.create("https://repo.grdev.net/artifactory/public")
+    }
 }
 
 val mvnExtension: Configuration by configurations.creating { isCanBeConsumed = true; isCanBeResolved = false }
 
 dependencies {
     compileOnly("org.apache.maven:maven-core:3.8.7")
-    compileOnly("com.gradle:gradle-enterprise-maven-extension:${gradleEnterpriseMavenExtensionVersion}")
+    compileOnly("com.gradle:develocity-maven-extension:${develocityMavenExtensionVersion}")
```

Comment:
❗ We need a copy of https://github.com/jenkinsci/gradle-plugin/blob/master/configuration-maven-extension/src/main/java/com/gradle/jenkins/maven/extension/internal/DevelocityConfigurerListener.java#L14 which will implement `DevelocityListener`. Like in CCUD: [GE listener](https://github.com/gradle/common-custom-user-data-maven-extension/blob/main/src/main/java/com/gradle/CommonCustomUserDataGradleEnterpriseListener.java) and [DV listener](https://github.com/gradle/common-custom-user-data-maven-extension/blob/main/src/main/java/com/gradle/CommonCustomUserDataDevelocityListener.java). 

I think it is better to wait with this change until https://github.com/gradle/develocity-agent-adapters is released.